### PR TITLE
test: Increase unit test code coverage in internal/kafka/internal/metrics

### PR DESCRIPTION
**Pull request by VanillaSpoon**
## Description
[MGDSTRM-7978](https://issues.redhat.com/browse/MGDSTRM-7978)

## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- [x] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] ~~Documentation added for the feature~~
- [ ] CI and all relevant tests are passing
- [x] Code Review completed
- [x] Verified independently by reviewer
- [ ] ~~Required metrics/dashboards/alerts have been added (or PR created).~~
- [ ] ~~Required Standard Operating Procedure (SOP) is added.~~
- [ ] ~~JIRA has created for changes required on the client side~~

### CODE REVIEW

File: internal/kafka/internal/metrics/component_versions_test.go [hunk 1]
```diff
@@ -59,3 +59,50 @@ func TestVersionsMetrics_Collect(t *testing.T) {
...
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m := newVersionMetrics(tt.fields.kafkaService)
+			ch := tt.args.ch
+			m.Describe(ch)
+			if len(ch) != tt.want {
+				t.Errorf("expect to have %d metrics but got %d", tt.want, len(ch))
+			}
```
Comment (machi1990) [inline internal/kafka/internal/metrics/component_versions_test.go:105]: let's try to use gomega assertions here. Just like https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/019ed9a849f792938f7a2b442f2ddc0164cb1e98/pkg/auth/context_test.go#L70 for example
```suggestion
			Expect(tt.want).To(Equal(len(ch)))
```
or much better alternative
```suggestion
			Expect(ch).To(HaveLen(tt.want))
```

Comment (VanillaSpoon) [reply internal/kafka/internal/metrics/component_versions_test.go]: Few other changes to make actually :)

**Comment by machi1990** [comment]
What's the difference of coverage between the before/after this change?

**Comment by VanillaSpoon** [comment]
**Before the change:**
___
- component_versions.go = 78.9
- federate_user_metrics.go = 94.4
- providers.go = 0

**After the change:** (Provided the tests are correctly implemented)
___
- component_versions.go = 89.5
- federate_user_metrics.go = 94.4
- providers.go = ignored from coverage

@machi1990

**Comment by machi1990** [comment]
Thanks @VanillaSpoon for the numbers

I'll leave it to @pawelpaszki for a fully review.

File: internal/kafka/internal/metrics/component_versions_test.go [hunk 2]
```diff
@@ -44,18 +45,63 @@ func TestVersionsMetrics_Collect(t *testing.T) {
...
+		},
+	}
+
+	for _, tt := range tests {
+		RegisterTestingT(t)
+		t.Run(tt.name, func(t *testing.T) {
+			m := newVersionMetrics(tt.fields.kafkaService)
+			ch := tt.args.ch
+			m.Describe(ch)
+			Expect(ch).To(HaveLen(tt.want))
```
Comment (pawelpaszki) [inline internal/kafka/internal/metrics/component_versions_test.go:104]:
```suggestion
	RegisterTestingT(t)
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			m := newVersionMetrics(tt.fields.kafkaService)
			ch := tt.args.ch
			m.Describe(ch)
			Expect(ch).To(HaveLen(tt.want))
```

File: internal/kafka/internal/metrics/component_versions_test.go [hunk 3]
```diff
@@ -44,18 +45,63 @@ func TestVersionsMetrics_Collect(t *testing.T) {
 				},
 			}},
 			args: args{ch: make(chan prometheus.Metric, 100)},
-			want: 8,
+			want: 9,
 		},
 	}
 
 	for _, tt := range tests {
+		RegisterTestingT(t)
```
Comment (pawelpaszki) [inline internal/kafka/internal/metrics/component_versions_test.go:53]: please move this line above `for _, tt := range tests {`

**Comment by pawelpaszki** [comment]
two minor comments. otherwise LGTM

File: internal/kafka/internal/metrics/component_versions_test.go (hunk 3 above)
Comment (VanillaSpoon) [reply internal/kafka/internal/metrics/component_versions_test.go]: Got it, thanks @pawelpaszki

**Review by pawelpaszki — APPROVED** [review]

**Pull request merged**